lib/bolt/cli.rb

```diff
@@ -477,38 +476,25 @@ def outputter
       @outputter ||= Bolt::Outputter.for_format(@config[:format])
     end
 
-    def load_task_data(name, modulepath)
-      module_name, file_name = name.split('::', 2)
-      file_name ||= 'init'
-
-      begin
-        env = Puppet::Node::Environment.create('bolt', modulepath)
-        Puppet.override(environments: Puppet::Environments::Static.new(env)) do
-          data = Puppet::InfoService::TaskInformationService.task_data(
-            env.name, module_name, name
-          )
-
-          file = data[:files].find { |f| File.basename(f, '.*') == file_name }
-          if file.nil?
-            raise Bolt::CLIError, "Failed to load task file for '#{name}'"
+    def load_task(name, args, modulepath)
+      task_signature = nil
+      error = nil
+      Puppet.initialize_settings
+      Puppet::Pal.in_tmp_environment('bolt', modulepath: [BOLTLIB_PATH] + modulepath, facts: {}) do |pal|
+        pal.with_script_compiler do |compiler|
```

Comment:
The `#run_task` command should be called within script_compiler scope, or it will not find the loader or script_compiler. That's partly why I think this approach is wrong. If the loading of Task is done outside the function, then the compiler needs to be set up twice.